receiver/awscloudwatchreceiver/logs.go

```diff
@@ -0,0 +1,353 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awscloudwatchreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscloudwatchreceiver"
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+)
+
+type logsReceiver struct {
+	region              string
+	profile             string
+	imdsEndpoint        string
+	pollInterval        time.Duration
+	maxEventsPerRequest int
+	namedPolls          []groupNames
+	prefixedPolls       []groupPrefix
```

Comment:
The naming in this file needs to be cleaned up. We're referring to "groups" as "polls", a single "group + streams" as "groups", etc.


I've made some suggestions, but to summarize - each instance of what you are currently calling the `pollConfig` interface is more easily understood as a request for a group's events, or a `groupRequest`. You can use this interface to generalize the way you're handling groups, not just the way you're making requests. 

What you call `groupNames` is actually a set of streamNames within a group. Just take the group name out and make this an alias. Pass the group name into the request function. Similar changes for `groupPrefix`.

```suggestion
	group          map[string]groupRequest
```

Reply:
I think one problem about this approach is that it assumes that each log group will only have one request. However the way the request struct is structured we'd need the capability to request per prefix and per list of fully qualified names. I'm fine with the naming changes however I don't think we can have the map looping logic you are suggesting for this reason.